src/prefect/settings.py

```diff
@@ -1357,12 +1393,20 @@ def default_cloud_ui_url(settings, value):
 Number of missed polls before a runner is considered unhealthy by its webserver.
 """
 
-PREFECT_RUNNER_SERVER_HOST = Setting(str, default="0.0.0.0")
+PREFECT_RUNNER_SERVER_HOST = Setting(
+    str,
+    default="0.0.0.0",
+    value_callback=check_for_deprecated_runner_server_host,
```

Comment:
Is it the case that `value_callback` will alway runs? Even if the setting isn't explicitly set and is taking on its default value?

Reply:
I believe it will only run when the deprecated setting is explicitly set - will verify this